Controller/Hpp/ReturnAction.php

```
@@ -0,0 +1,176 @@
+use NexiCheckout\Model\Result\RetrievePaymentResult;
+use Psr\Log\LoggerInterface;
+use Nexi\Checkout\Gateway\Http\Client;
+
+class ReturnAction implements ActionInterface
+{
+    /**
+     * @param RedirectFactory $resultRedirectFactory
+     * @param RequestInterface $request
+     * @param UrlInterface $url
+     * @param Session $checkoutSession
+     * @param Config $config
+     * @param Builder $transactionBuilder
+     * @param OrderRepository $orderRepository
+     * @param LoggerInterface $logger
+     * @param ManagerInterface $messageManager
+     * @param Client $client
+     */
+    public function __construct(
+        private readonly RedirectFactory  $resultRedirectFactory,
+        private readonly RequestInterface $request,
+        private readonly UrlInterface     $url,
+        private readonly Session          $checkoutSession,
+        private readonly Config           $config,
+        private readonly Builder          $transactionBuilder,
+        private readonly OrderRepository  $orderRepository,
+        private readonly LoggerInterface  $logger,
+        private readonly ManagerInterface $messageManager,
+        private readonly Client           $client,
+    ) {
+    }
+
+    /**
+     * Execute action based on request and return result
+     *
+     * @return ResultInterface
+     */
+    public function execute(): ResultInterface
+    {
+        $order = $this->checkoutSession->getLastRealOrder();
+
+        $this->logger->debug(
+            'ReturnAction request: ' . json_encode($this->request->getParams())
+            . ' - Order ID: ' . $order->getIncrementId()
+            . 'http referrer: ' . $this->request->getServer('HTTP_REFERER')
+        );
+
+        try {
+            if ($order->getPayment()->getAdditionalInformation('payment_id') != $this->request->getParam('paymentid')) {
+                throw new LocalizedException(__('Payment ID does not match.'));
```

what is the point of using LocalizedException if you are doing try catch all Exceptions are catch and this message is not presented to user? 
maybe you should not catch general Exception but expect what can fail in this code and catch only specific exceptions